corehq/apps/sms/templates/sms/add_gateway.html

```diff
@@ -5,22 +5,14 @@
 
 {% block js %}{{ block.super }}
     <script src="{% static 'sms/js/add_gateway_form_handler.ko.js' %}"></script>
-{% endblock %}
-
-{% block js-inline %}{{ block.super }}
-    <script>
-        var gatewayFormHandler = new AddGatewayFormHandler({
-            share_backend: {{ form.give_other_domains_access.value|BOOL }},
-            use_load_balancing: {{ use_load_balancing|BOOL }},
-            phone_numbers: {{ form.phone_numbers.value|default:'[]'|safe }},
-            phone_number_required_text: "{% trans 'You must have at least one phone number.' %}"
-        });
-        $('#add-gateway-form').koApplyBindings(gatewayFormHandler);
-        gatewayFormHandler.init();
-    </script>
+    <script src="{% static 'sms/js/add_gateway.js' %}"></script>
 {% endblock %}
 
 {% block page_content %}
+    {% initial_page_data 'form.give_other_domains_access' form.give_other_domains_access.value|BOOL %}
```

Comment:
Nitpick: initial_page_data does a json conversion, so the `BOOL` shouldn't be needed.

Reply:
Unrelated, `form.give_other_domains_access` is kind of a weird naming convention. From the javascript's perspective, this is just a flag, it doesn't matter than it came from a form.